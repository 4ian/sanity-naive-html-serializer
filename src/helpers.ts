import sanityClient from 'part:@sanity/base/client'
import { SanityDocument } from '@sanity/types'

const client = sanityClient.withConfig({ apiVersion: '2021-03-25' })

export const findLatestDraft = (documentId: string, ignoreI18n = true) => {
  //eliminates i18n versions
  const query = `*[_id match $id ${
    ignoreI18n ? ' && (_id in path("drafts.*") || _id in path("*"))' : ''
  }]`
  const params = { id: `*${documentId}` }
  return client
    .fetch(query, params)
    .then(
      (docs: SanityDocument[]) =>
        docs.find(doc => doc._id.includes('draft')) ?? docs[0]
    )
}
