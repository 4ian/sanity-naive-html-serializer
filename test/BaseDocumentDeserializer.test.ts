import { getDeserialized } from './helpers'
const documentLevelArticle = require('./__fixtures__/documentLevelArticle')
const fieldLevelArticle = require('./__fixtures__/fieldLevelArticle')

let mockTestKey = 0

jest.mock('@sanity/block-tools/lib/util/randomKey.js', () => {
  return jest.fn().mockImplementation(() => {
    return `randomKey-${++mockTestKey}`
  })
})

beforeEach(() => {
  mockTestKey = 0
})

test('Global test of working doc-level functionality and snapshot match', () => {
  const deserialized = getDeserialized(documentLevelArticle, 'document')
  expect(deserialized).toMatchSnapshot()
})

test('Global test of working field-level functionality and snapshot match', () => {
  const deserialized = getDeserialized(fieldLevelArticle, 'field')
  expect(deserialized).toMatchSnapshot()
})

test('Contains id of original document', () => {
  const deserialized = getDeserialized(documentLevelArticle, 'document')
  const id = deserialized._id
  expect(id).toEqual(documentLevelArticle._id)
})

test('Contains rev of original document', () => {
  const deserialized = getDeserialized(documentLevelArticle, 'document')
  const rev = deserialized._rev
  expect(rev).toEqual(documentLevelArticle._rev)
})

test('Contains type of original document', () => {
  const deserialized = getDeserialized(documentLevelArticle, 'document')
  const type = deserialized._type
  expect(type).toEqual(documentLevelArticle._type)
})
