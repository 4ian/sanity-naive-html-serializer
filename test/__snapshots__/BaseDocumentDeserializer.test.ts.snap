// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global test of working doc-level functionality and snapshot match 1`] = `
Object {
  "config": Object {
    "nestedArrayField": Array [
      Object {
        "_key": "4a58adc7c507",
        "_type": "block",
        "children": Array [
          Object {
            "_key": "19bb3419e1660",
            "_type": "span",
            "marks": Array [],
            "text": "This is block text 1 level deep",
          },
        ],
        "markDefs": Array [],
        "style": "normal",
      },
    ],
    "objectAsField": Array [
      Object {
        "_key": "mrqq84",
        "_type": "childObjectField",
        "content": Object {
          "": Object {
            "_key": "1aa228ac848b",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "7a49e74b2a490",
                "_type": "span",
                "marks": Array [],
                "text": "This is a block text 2 levels deep",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
        },
        "title": "This is one level deeper",
      },
    ],
    "title": "This is an object nested in a document",
  },
  "content": Object {
    "": Object {
      "_key": "a4d177e92666",
      "_type": "block",
      "children": Array [
        Object {
          "_key": "d11fab010f4d0",
          "_type": "span",
          "marks": Array [],
          "text": "This is block text at the top level.",
        },
      ],
      "markDefs": Array [],
      "style": "normal",
    },
    "objectField": Object {
      "objectAsField": Object {
        "childObjectField": Object {
          "content": Object {
            "": Object {
              "_key": "4f24f6fbfae7",
              "_type": "block",
              "children": Array [
                Object {
                  "_key": "4da3288854960",
                  "_type": "span",
                  "marks": Array [],
                  "text": "This is block text in a nested object in an object in top-level block text.",
                },
              ],
              "markDefs": Array [],
              "style": "normal",
            },
          },
          "title": "This is a nested object in an object in top-level block text.",
        },
      },
      "title": "This is an object in top-level block text.",
    },
  },
  "hidden": "",
  "snippet": "This is text in my text field.",
  "tags": Object {
    "": "tag 3",
  },
  "title": "My Document-Level Article",
}
`;
